File: php/examples/get_invoices.php
<?php
require_once 'base_request.php';

// Example of GET /invoices
// An array of company names or company_ids may be provided, but not both

$debug = true;
$company_id   = $_ENV['COMPANY_ID'];
$request_data = '';
$request_params = ['company_ids' => [$company_id]];

send_request('GET', '/invoices', $request_params, $request_data);

File: csharp/examples/GetInvoices.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Net.Http;
namespace Examples.examples
{
    // Example of GET /invoices
    
    // An array of company names or company_ids may be provided, but not both

    public class GetInvoices : BaseRequest
	{
        private string _companyIds;

        public GetInvoices()
		{
            dotenv.net.DotEnv.Load();
            _companyIds = Environment.GetEnvironmentVariable("COMPANY_ID");
            Console.WriteLine($"GetInvoices: _companyIds={_companyIds}");
        }

        public override string SendRequest()
         {
            // Construct the url with the ids param as an array in the query string
            var queryParams = new List<string>();
            foreach (var id in _companyIds.Split(','))
            {
                queryParams.Add($"company_ids[]={Uri.EscapeDataString(id)}");
            }
            var queryString = string.Join("&", queryParams);
            var urlWithParams = $"invoices?{queryString}";
            return GetRequest(urlWithParams);
         }
    }
}

